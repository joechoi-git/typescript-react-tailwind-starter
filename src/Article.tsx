import React from "react";

interface ArticleProps {
    title: string;
    content: string;
}

export default function Article({ title, content }: ArticleProps): React.JSX.Element {
    return (
        <>
            <header>
                <h1>{title}</h1>
                <p>
                    Published on <time dateTime="2024-06-28">June 28, 2024</time> by{" "}
                    <span>Author Name</span>
                </p>
            </header>
            <article>
                <section>
                    <p>{content}</p>
                </section>
            </article>
        </>
    );
}
