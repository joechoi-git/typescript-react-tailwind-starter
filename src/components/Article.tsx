import React from "react";
import { getCurrentDate } from "../utils/helper";

export interface ArticleProps {
    title: string;
    content: string;
}

export default function Article({ title, content }: ArticleProps): React.JSX.Element {
    return (
        <>
            <header>
                <h1 className="text-2xl">{title}</h1>
                <p>
                    Published on <time dateTime="2024-06-28">{getCurrentDate()}</time>
                </p>
            </header>
            <article>
                <section>
                    <p>{content}</p>
                </section>
            </article>
        </>
    );
}
