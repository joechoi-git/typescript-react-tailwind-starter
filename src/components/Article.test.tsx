import React from "react";
import { render, screen } from "@testing-library/react";
import Article, { ArticleProps } from "./Article";

describe("Article Component", () => {
    const articleProps: ArticleProps = {
        title: "Example Article Title",
        content: "Example content of the article."
    };

    test("renders article with props correctly", () => {
        render(<Article {...articleProps} />);

        // Assert that the article title and content are rendered correctly
        expect(screen.getByText(articleProps.title)).toBeInTheDocument();
        expect(screen.getByText(articleProps.content)).toBeInTheDocument();
    });

    test("matches snapshot", () => {
        const { asFragment } = render(<Article {...articleProps} />);

        // Use toMatchSnapshot to compare the rendered output with a stored snapshot
        expect(asFragment()).toMatchSnapshot();
    });
});
