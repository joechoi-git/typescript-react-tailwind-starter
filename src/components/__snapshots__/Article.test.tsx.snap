// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Example Article Title
    </h1>
    <p>
      Published on 
      <time
        datetime="2024-06-28"
      >
        June 28, 2024
      </time>
       by 
      <span>
        Author Name
      </span>
    </p>
  </header>
  <article>
    <section>
      <p
        class="text-blue-600 dark:text-white"
      >
        Example content of the article.
      </p>
    </section>
  </article>
</DocumentFragment>
`;
